getting-started/setup-building.rst

```diff
@@ -458,6 +458,93 @@ used in ``python.sh``:
 .. _wasmtime: https://wasmtime.dev
 .. _WebAssembly: https://webassembly.org
 
+
+Emscripten
+----------
+
+Emscripten_ is a complete open source compiler toolchain. It compiles C/C++ code
+into WebAssembly_/JavaScript executables, for use in JavaScript runtimes,
+including browsers and Node.js.
+
+.. note::
+
+   The instructions below assume a Unix-based OS due to cross-compilation for
+   CPython being designed for ``./configure`` / ``make``.
+
+To build for Emscripten, you will need to cross-compile CPython. This requires a
+C compiler just like building for :ref:`Unix <unix-compiling>` as well as:
+
+1. The Emscripten compiler
+2. Node.js
+
+The simplest way to install the Emscripten compiler is as follows:
+
+.. code-block:: sh
+
+    # Install Emscripten
+    git clone emscripten/emsdk
+    ./emsdk/emsdk install 4.0.5
+    ./emsdk/emsdk activate 4.0.5
+    source ./emsdk/emsdk_env.sh
+
+Building for Emscripten requires doing a cross-build where you have a *build*
+Python to help produce a Emscripten build of CPython. This means you build
+CPython twice: once to have a version of Python for the build system to use and
+another that's the build you ultimately care about (that is, the build Python is
+not meant for use by you directly, only the build system).
+
+The easiest way to get a debug build of CPython for Emscripten is to use the
+``Tools/wasm/emscripten build`` command (which should be run with a recent
+version of Python you have installed on your machine):
+
+.. code-block:: shell
+
+   $ python3 Tools/wasm/emscripten build --quiet -- --config-cache --with-pydebug
```

Comment:
Only need the prompt for `console` code-blocks and when showing output (so it uses the right syntax highlighting):
```suggestion
.. code-block:: shell

   python3 Tools/wasm/emscripten build --quiet -- --config-cache --with-pydebug
```